src/components/ens-profile/ActionButtons/WatchButton.tsx

```diff
@@ -0,0 +1,92 @@
+import ConditionalWrap from 'conditional-wrap';
+import lang from 'i18n-js';
+import debounce from 'lodash/debounce';
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { ContextMenuButton, MenuConfig } from 'react-native-ios-context-menu';
+import ActionButton from './ActionButton';
+import { useWatchWallet } from '@rainbow-me/hooks';
+
+export default function WatchButton({
+  address,
+  ensName,
+  avatarUrl,
+}: {
+  address?: string;
+  ensName?: string;
+  avatarUrl?: string | null;
+}) {
+  const { isWatching, watchWallet } = useWatchWallet({
+    address,
+    avatarUrl,
+    ensName,
+    showImportModal: false,
+  });
+
+  // An "optimistic" state will provide us with optimistic feedback on the UI,
+  // and not wait for the import to finish.
+  const [optimisticIsWatching, setOptimisticIsWatching] = useState(isWatching);
+
+  // We need a ref here to avoid recreating the debounce function.
+  const optimisticIsWatchingRef = useRef<boolean>();
+  optimisticIsWatchingRef.current = optimisticIsWatching;
+
+  const handleWatch = useCallback(() => {
+    if (isWatching !== optimisticIsWatchingRef.current) {
+      watchWallet();
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+  // We want to debounce the watch/unwatch functionality to avoid spamming the
+  // JS thread.
+  const debouncedWatchWallet = debounce(handleWatch, 500);
+
+  const handlePressWatch = useCallback(() => {
+    debouncedWatchWallet();
```

Comment:
here you additionally rely on that the function will not change. But it might. We should consider deps array as a suggestion for React.

Reply:
This has also been removed in https://github.com/rainbow-me/rainbow/pull/3418

Reply:
Can we cherry-pick this? This is really naughty

Reply:
yepperss